ls1intum/Artemis
# `Programming exercises`: Fix building animation for first submission

### Checklist
#### General


#### Client

### Motivation and Context


Fix an issue where student did not see the "building and testing" animation during their first push.

### Description

This PR changes the behavior of the result in the exercise detail view for programming exercises to show "No result" after starting a programming exercise but before the due date if there is currently no result to display. Before that "No submission" was shown. This new behavior should be more in line with the behavior before this PR: #5896

This change now allows the result to react to the first submission to the vcs as well and therefore display the building animation once again for the first push.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Programming Exercise
- 1 Modeling (Text and File upload should behave the same) Exercise
- 1 Quiz Exercise

1. Log in to Artemis
2. Navigate to Course Administration
3. Check that the building animation is now once again working for the first push in programming exercises
4. Check that the lifecycle for the rest of the programming exercise is working. I.e. the result will always be displayed if it exists. Before starting the exercise or when missing the due date and without any result a helpful short text displays the correct state.
5. Check that modeling exercises still work and only display "No result"/the actual result when the student submitted.
6. Check that the quiz lifecycle correctly displays the submission state before, during and after the quiz time ended

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage




#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [submission-result-status.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5121-2/artifact/shared/Coverage-Report-Client-Tests/app/overview/submission-result-status.component.ts.html) | 100% | ✅ |

### Screenshots


![Bildschirmfoto 2023-07-06 um 13 12 58](https://github.com/ls1intum/Artemis/assets/38322605/fbe2ed9b-cd05-4306-9cdd-2855535b9b90)
